src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -496,19 +513,44 @@ public void LoadEnvironmentVariables()
 
         public string LogMode => this.Log.Mode is null ? this.Defaults.LogMode : this.Log.Mode;
 
-        // TODO
+        // TODO - Extensions
         XmlNode? IWinSWConfiguration.ExtensionsConfiguration => throw new NotImplementedException();
 
-        public List<string> ExtensionIds => throw new NotImplementedException();
+        public List<string> ExtensionIds => this.YamlExtensionIds ?? this.Defaults.ExtensionIds;
 
-        public string BaseName => throw new NotImplementedException();
+        public string BaseName => this.Defaults.BaseName;
 
-        public string BasePath => throw new NotImplementedException();
+        public string BasePath => this.Defaults.BasePath;
 
-        public string? ServiceAccountDomain => throw new NotImplementedException();
+        public string? SecurityDescriptor => this.SecurityDescriptorYaml ?? this.Defaults.SecurityDescriptor;
 
-        public string? ServiceAccountName => throw new NotImplementedException();
+        public static TimeSpan ParseTimeSpan(string v)
```

Comment:
Code duplication from XML? I would suggest to move it to a separate utility class then

Reply:
Make sense. will update.